web-app/js/smartR/_angular/controllers/heatmap.js

```diff
@@ -8,179 +8,115 @@ function HeatmapController ($scope, smartRUtils, commonWorkflowService) {
         },
-        params :  {
-            aggregateProbes : false
+        params:  {
+            aggregateProbes: false
         },
-        scriptResults : {}
+        running: false,
+        scriptResults: {}
     };
 
     // ------------------------------------------------------------- //
     // Run Heatmap                                                   //
     // ------------------------------------------------------------- //
     $scope.runAnalysis = {
-        btn : {
-            disabled : true
+        btn: {
+            disabled: false
+        },
+        tab: {
+            disabled: true
         },
         params: {
-            max_row : 100,
-            sorting : 'nodes',
-            ranking : 'coef'
+            max_row: 100,
+            sorting: 'nodes',
+            ranking: 'coef'
         },
-        download : {
-            disabled : true
+        download: {
+            disabled: true
         },
-        subsets : 0,
-        scriptResults : {}
+        running: false,
+        subsets: 0,
+        scriptResults: {}
     };
 
     commonWorkflowService.registerCondition(
-        ['fetch.conceptBoxes.highDimensional'],
-        ['fetch.btn'],
-        this.whenSelectHighDimensionalNodes
-    );
-
-    commonWorkflowService.registerCondition(
-        ['fetch.btn.disabled', 'fetch.totalSamples', 'fetch.subsets'],
-        ['preprocess', 'runAnalysis'],
-        this.whenFetchHeatmapData
-    );
-
-    commonWorkflowService.registerCondition(
-        ['preprocess.btn.disabled'],
-        ['fetch', 'runAnalysis'],
-        this.whenPreprocessHeatmapData
-    );
-
-    commonWorkflowService.registerCondition(
-        ['runAnalysis.btn.disabled', 'runAnalysis.scriptResults'],
-        ['fetch', 'preprocess', 'runAnalysis.download'],
-        this.whenRunHeatmapAnalysis
+        ['fetch.running',
+            'preprocess.running',
+            'runAnalysis.running',
+            'fetch.conceptBoxes.highDimensional.valid'
+        ],
+        ['fetch', 'preprocess', 'runAnalysis'],
+        function(newValues, oldValues, scope, affectedComponents) {
+            var fetchRunning = newValues[0],
+                preprocessRunning = newValues[1],
+                runAnalysisRunning = newValues[2],
+                conceptBoxesHighDimensionalValid = newValues[3];
+            var fetchModel = affectedComponents[0],
+                preprocessModel = affectedComponents[1],
+                runAnalysisModel = affectedComponents[2];
+
+            // clear old results
+            if (fetchRunning) {
+                preprocessModel.scriptResults = {};
+                runAnalysisModel.scriptResults = {};
+            }
+
+            // clear old results
+            if (preprocessRunning) {
+                runAnalysisModel.scriptResults = {};
+            }
+
+            // disable tabs when certain criteria are not met
+            fetchModel.tab.disabled = preprocessRunning || runAnalysisRunning;
```

Comment:
I would not recommend this. We have input from users that they still wanna be able to view and configure options for next process while something is running in the background. For example, while fetching the data (which is normally long) I want to be able to prepare options.